custom-vu/src/main/kotlin/jces1209/vu/action/SearchCloudFilter.kt

```diff
@@ -0,0 +1,26 @@
+package jces1209.vu.action
+
+import com.atlassian.performance.tools.jiraactions.api.SEARCH_WITH_JQL
+import com.atlassian.performance.tools.jiraactions.api.WebJira
+import com.atlassian.performance.tools.jiraactions.api.action.Action
+import com.atlassian.performance.tools.jiraactions.api.measure.ActionMeter
+import com.atlassian.performance.tools.jiraactions.api.memories.Memory
+import jces1209.vu.page.CloudIssueNavigator
+import java.net.URI
+
+class SearchCloudFilter(
+    private val jira: WebJira,
+    private val meter: ActionMeter,
+    private val filters: Memory<URI>
+) : Action {
+    override fun run() {
+        val filter = filters.recall()!!
+        meter.measure(
+            key = SEARCH_WITH_JQL,
+            action = {
+                jira.navigateTo(filter.toString())
+                CloudIssueNavigator(jira.driver).waitForNavigator()
```

Comment:
This is second use I see of `CloudIssueNavigator` that is creating new instance but not like "using it", in such cases would it make sense to maybe make waitForNavigator like a companion/utility method or smth?
`CloudIssueNavigator.waitForNavigator(jira.driver)` ?

Reply:
Isn't waiting for it to load considered "using it"?

> companion/utility method or smth?

Why make things static/global when we can make them local? If we can pass page objects around, we can build relations, polymorphism, state passing. e.g. we could encapsulate `IssueNavigationFeature` as an interface, which would dispatch to Cloud/Server implementations, etc.

Also local stuff can hold local state, like caches, which would get GCd after becoming unreachable, whereas static caches would leak memory.